solidity/test/Polymer7683.t.sol

```diff
@@ -0,0 +1,440 @@
+        uint32 chainId,
+        address emittingContract,
+        bytes memory topics,
+        bytes memory unindexedData
+    ) {
+        return (expectedChainId, expectedEmitter, expectedTopics, expectedData);
+    }
+
+    function inspectLogIdentifier(bytes calldata proof) external pure returns (uint32 srcChain, uint64 blockNumber, uint16 receiptIndex, uint8 logIndex) {
+        return (0, 0, 0, 0);
+    }
+
+    function inspectPolymerState(bytes calldata proof) external pure returns (bytes32 stateRoot, uint64 height, bytes memory signature) {
+        return (bytes32(0), 0, "");
+    }
+}
+
+contract Polymer7683Test is Test {
+    using TypeCasts for address;
+
+    Polymer7683 polymer7683;
+    MockCrossL2Prover prover;
+    
+    address owner;
+    address permit2;
+    uint256 localChainId;
+    uint256 destChainId;
+    address destContract;
+
+    function setUp() public {
+        owner = makeAddr("owner");
+        permit2 = makeAddr("permit2");
+        localChainId = 1;
+        destChainId = 2;
+        destContract = makeAddr("destContract");
+
+        prover = new MockCrossL2Prover();
+        
+        vm.prank(owner);
+        polymer7683 = new Polymer7683(
+            ICrossL2ProverV2(address(prover)),
+            permit2,
+            localChainId
+        );
+    }
+
+    function _createSettlementProof(
+        uint256 chainId,
+        address emitter,
+        bytes32 orderId,
+        bytes memory fillerData
+    ) internal returns (bytes memory) {
+        OrderData memory orderData = OrderData({
+            sender: TypeCasts.addressToBytes32(address(this)),
+            recipient: TypeCasts.addressToBytes32(address(this)),
+            inputToken: bytes32(0),
+            outputToken: bytes32(0),
+            amountIn: 1e18,
+            amountOut: 2e18,
+            senderNonce: 0,
+            originDomain: uint32(localChainId),
+            destinationDomain: uint32(chainId),
+            destinationSettler: TypeCasts.addressToBytes32(emitter),
+            fillDeadline: uint32(block.timestamp + 1 hours),
+            data: ""
+            });
+
+        bytes memory originData = OrderEncoder.encode(orderData);
+    
+        // Create the proof data
+        bytes memory topics = abi.encodePacked(
+            keccak256("Filled(bytes32,bytes,bytes)"),
+            bytes32(chainId)
+        );
+    
+        bytes memory data = abi.encode(orderId, originData, fillerData);
+    
+        prover.setExpectedEvent(
+            uint32(chainId),
+            emitter,
+            topics,
+            data
+        );
+    
+        return "dummy_proof";
+    }
+
+    function _createRefundProof(
+        uint256 chainId,
+        address emitter,
+        bytes32[] memory orderIds
+    ) internal returns (bytes memory) {
+        bytes memory topics = abi.encodePacked(
+            keccak256("Refunded(bytes32,address)"),
+            bytes32(chainId)
+        );
```

Comment:
This is using the wrong event. The previous one was the right one.
`Refunded` y emitted in the in the order's origin chain once the oder's tokens are sent to the user, and `Refund` is emitted on the order's destination chain when the user triggers the refund, so `Refund` event proof is what you need to check for.

Reply:
good catch, fixed!